packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -2040,6 +2068,25 @@ class _VerticalOuterDimension extends Scrollable {
 class _VerticalOuterDimensionState extends ScrollableState {
   DiagonalDragBehavior get diagonalDragBehavior => (widget as _VerticalOuterDimension).diagonalDragBehavior;
 
+  // Implemented in the _HorizontalInnerDimension instead.
+  @override
+  (List<Future<void>>, ScrollableState) _performEnsureVisible(
+    RenderObject object, {
+    double alignment = 0.0,
+    Duration duration = Duration.zero,
+    Curve curve = Curves.ease,
+    ScrollPositionAlignmentPolicy alignmentPolicy = ScrollPositionAlignmentPolicy.explicit,
+    RenderObject? targetRenderObject,
+  }) {
+    assert(
+      false,
```

Comment:
Can you explain a bit why this won't be called? I thought the ensureVisible in Scrollable will recursively find Scrollable and call this method. Wouldn't this be called after _HorizontalInnerDimension._performEnsureVisible is called?

Reply:
The recursive logic for bubbling up through 1D Scrollables is different because each scrollable has its own viewport. As it iterates through each one, the next scrollable context is used to find the next viewport.
In 2D scrolling, there are two scrollables but only one viewport. The horizontal one will always be the inner one that is reached first, so it handles it for each scrollable and then returns the vertical scrollable to continue to original logic up through thew nest scrollable/viewport pair.

Reply:
Isn't the code using `Scrollable.maybeOf` which would be grabbing _ScrollableScope where both horizontal and vertical will create?

Reply:
Yup! It's after that where we use the context to find the next scrollable above that where it breaks for 2D. This is why I split out the implementations between _performEnsureVisible. Originally I had a `is` check in https://github.com/flutter/flutter/pull/128812/files/575eb0dd73ed49194fd3e7b6556c67a4092798b5 to handle 2D as a special case, which was less ideal.